### PR TITLE
feat: add conf_path in PetrelBackend

### PR DESCRIPTION
**Pull request by sunyc11**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Add conf_path in PetrelBackend initial parameters to set user-defined petrel client config path.

## Modification

Add conf_path in PetrelBackend initial parameters to set user-defined petrel client config path.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

**Before PR**:

- [ ] I have read and followed the workflow indicated in the [CONTRIBUTING.md](https://github.com/open-mmlab/mmcv/blob/master/CONTRIBUTING.md) to create this PR.
- [ ] Pre-commit or linting tools indicated in [CONTRIBUTING.md](https://github.com/open-mmlab/mmcv/blob/master/CONTRIBUTING.md) are used to fix the potential lint issues.
- [ ] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] New functionalities are covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [ ] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [ ] If the modification has potential influence on downstream or other related projects, this PR should be tested with some of those projects, like MMDet or MMCls.
- [ ] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: mmcv/fileio/file_client.py [hunk 1]
```diff
@@ -108,14 +109,15 @@ class PetrelBackend(BaseStorageBackend):
 
     def __init__(self,
                  path_mapping: Optional[dict] = None,
+                 conf_path: str = None,
```
Comment (zhouzaida) [inline mmcv/fileio/file_client.py:112]: Adding a new parameter in the middle position may cause a bc issue. Suggest moving it to the end.

File: mmcv/fileio/file_client.py [hunk 2]
```diff
@@ -95,6 +95,7 @@ class PetrelBackend(BaseStorageBackend):
         path_mapping (dict, optional): Path mapping dict from local path to
             Petrel path. When ``path_mapping={'src': 'dst'}``, ``src`` in
             ``filepath`` will be replaced by ``dst``. Default: None.
+        conf_path (str, optional): Petrel client config path. Default: None.
```
Comment (zhouzaida) [inline mmcv/fileio/file_client.py:98]:
```suggestion
        conf_path (str, optional): Config path of Petrel client. Default: None.
            `New in version 1.7.1`.
```

Comment (zhouzaida) [reply mmcv/fileio/file_client.py:98]: move this line to the end.

**Comment by zhouzaida** [comment]
> [CLA assistant check](https://cla-assistant.io/open-mmlab/mmcv?pullRequest=2459) Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/open-mmlab/mmcv?pullRequest=2459) before we can accept your contribution.
> 
> **sunyecheng** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).
> You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/open-mmlab/mmcv?pullRequest=2459) it.

Hi, thanks for your contribution. Please sign the CLA before we merge this PR.

**Comment by sunyc11** [comment]
> 

hi, @zhouzaida , I've signed the CLA and adjusted the order of parameters.

**Review by zhouzaida — APPROVED** [review]

**Pull request merged**